pallets/thea/src/lib.rs

```diff
@@ -503,13 +505,14 @@ pub mod pallet {
 		) -> Result<(), DispatchError> {
 			ensure!(beneficiary.len() <= 100, Error::<T>::BeneficiaryTooLong);
 			// TODO: This will be refactored when work on withdrawal so not fixing clippy suggestion
-			let (network, ..) = asset_handler::pallet::Pallet::<T>::get_thea_assets(asset_id);
+			let (mut network, ..) = asset_handler::pallet::Pallet::<T>::get_thea_assets(asset_id);
+			if asset_id == T::PolkadexAssetId::get() {
+				network = 1;
+			};
```

Comment:
With this approach extra query to the storage will be executed always, but it's can be optional and executed only if `asset_id != T::PolkadexAssetId::get()`. So I think it's could be changed to something like this:

```rust
			let network = if asset_id == T::PolkadexAssetId::get() {
				1
			} else {
				let (network, ..) = asset_handler::pallet::Pallet::<T>::get_thea_assets(asset_id);
				network
			};

```